:construction_worker: Nomenclature and Testing :building_construction:

a5f9t4 opened the pull request:
closes #10

abigger87 committed:
fix nomenclature for erc20

abigger87 committed:
snake case cairo convention

abigger87 committed:
nomenclature

exp-table commented:
It leaves us with two questions ;
1 - Do we want to keep same naming convention as the current one used throughout web3 (i.e. camelcase - used everywhere in EVM) to not break "compatiblity" and allow develops to move to starknet (at least frontend-wise) with the minimum amount of modifications needed?
2 - If we do ; should the convention be the same throughout all the contracts, and not mix camel case and snake case?

exp-table committed:
Delete IERC721.cairo

OZ's implementation of the ERC721 isn't proper. Particularly because it has a concept of `allowance` which isn't part of the EIP-721.

exp-table committed:
Delete INERC721.cairo

Same reasons for IERC721.cairo

Commit:
Merge branch 'main' into nomenclature

abigger87 committed:
fix interfaces

abigger87 committed:
fix compile

abigger87 committed:
mock erc20

abigger87 committed:
mock erc20

abigger87 committed:
fix tests

abigger87 committed:
format

abigger87 committed:
tests + erc20 fix

abigger87 committed:
added a bunch of tests

exp-table commented on tests/test_ERC20.py:
```diff
@@ -1,41 +1,194 @@
+    assert executed_info.result.allowance == uint(0)
+
+@pytest.mark.asyncio
+async def test_increase_allowance_overflow():
```
For more clarity regarding the expected behavior of the test, the function should have the `test_fail` prefix imho

exp-table commented on tests/test_ERC20.py:
```diff
@@ -1,41 +1,194 @@
+    # Approve
+    await erc20.approve(friend.contract_address, amount).invoke(caller_address=owner.contract_address)
+    # Overflow
+    await assert_invoked_revert(erc20.increase_allowance(friend.contract_address, amount), owner.contract_address)
```
Not quite sure as I am not super familiar with pytest, but apparently the prefered method of expecting errors is

```
with pytest.raises(Exception):
    await erc20.increase_allowance(friend.contract_address, amount).invoke()
```

abigger87 committed:
:gear: test_fail prefix :test_tube:

abigger87 committed:
:gear: test_fail prefix :test_tube:

refcell committed:
:hammer: fix attributions :bug:
